cocotb/handle.py

```diff
@@ -664,8 +664,11 @@ def setimmediatevalue(self, value):
             TypeError: If target has an unsupported type for
                 real value assignment.
         """
-        if not isinstance(value, float):
-            self._log.critical("Unsupported type for real value assignment: %s (%s)", type(value), repr(value))
+        try:
+            value = float(value)
```

Comment:
This is probably a bad idea, it allows nonsense like `.value = "1.5"`

Reply:
That already slipped through the previous review round...

Reply:
One could put in a check like `isinstance(value, numbers.Number)` to catch all possible valid numeric values (including NumPy numeric types). However, these checks do start to get expensive if users put them in speed critical loops (and assigning values seems like an area where we should be careful about performance). A quick benchmark using `timeit` to compare some variations of this chunk of code.

1. This PR (`try: ... except:`): **220ns**
2. Original`isinstance(value, float)`: **250ns**
3. `isinstance(value, numbers.Number)`: **1,330ns**
4. `isinstance(value, float) or isinstance(value, int)`: **540ns**

I think letting some users do things like `.value = "1.5"`, although cringeworthy, is probably worth the 2-6x speed improvement we get from this as written. Some might even consider it a feature 😉.

Reply:
How does `isinstance(value, (float, int, numbers.Number))` compare?

Reply:
That one takes:
1. **515ns** if `value` is `float`.
2. **585ns** if `value` is `int`.
3. **1,580ns** if `value` is `np.uint8`.
4. **1,930ns** if `value` is `str`.

Reply:
Another option that should be both fast and correct is to use `value.__float__()`, with the same `try`  / `except` as you have now.

Reply:
That one is faster than the others at about **410ns**. I'm actually a bit surprised that it's not closer to the `float(value)` solution.